bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/GeneratorUtils.xtend

```diff
@@ -204,19 +205,7 @@ class GeneratorUtils {
 		val program = EcoreUtil2.getContainerOfType(event, Program);
 		if(program !== null) {
 			// count event handlers, so we get unique names
-			var occurence = 1;
-			var found = false;
-			for(e: program.eventHandlers) {
-				if(e.equals(event)){
-					found = true;
-				}
-				// no break; statement => need flag
-				// only count events with the same name
-				if(!found && e.baseName.equals(event.baseName)) {
-					occurence++;
-				}
-			}
-			return '''HandleEvery«event.baseName»«occurence»''';
+			return '''HandleEvery«event.baseName»_«(event.eContainer as Program).eventHandlers.filter[it.event.baseName == event.event.baseName].indexed.filter[it.value === event].head.key + 1»''';
```

Comment:
nit: I'd have pulled out the count computation to make the naming look more similar to the one below (we do just add a number here as compared to the other handler naming)

Reply:
```suggestion
                        val occurrence = (event.eContainer as Program).eventHandlers.filter[it.event.baseName == event.event.baseName].indexed.filter[it.value === event].head.key + 1;
			return '''HandleEvery«event.baseName»_«occurence»''';
```